### PR TITLE
fix: tidy compat and update versions

### PR DESCRIPTION
**Pull request by laurentsenta**
build for the `master` group started failing with the following error:

```
unused github.com/whyrusleeping/multiaddr-filter
unused golang.org/x/xerrors
go: downloading github.com/google/go-cmp v0.5.6
github.com/libp2p/test-plans/ping/go imports
        github.com/testground/sdk-go/sync imports
        nhooyr.io/websocket tested by
        nhooyr.io/websocket.test imports
        nhooyr.io/websocket/internal/test/assert imports
        github.com/google/go-cmp/cmp loaded from github.com/google/go-cmp@v0.5.5,
        but go 1.16 would select v0.5.6
...
```

This updates the go version (followup https://github.com/libp2p/test-plans/pull/23#discussion_r911688297)
and use `-compat` option in CI to prevent the selection error.

### CODE REVIEW

**Comment by laurentsenta** [comment]
(passing run: https://github.com/singulargarden/test-plans/runs/7193005032?check_suite_focus=true)

File: ping/_compositions/go-cross-versions.toml [hunk 1]
```diff
@@ -93,10 +97,14 @@
...
+    build_base_image = 'golang:1.18-buster'
     modfile = "go.v0.20.mod"
 
   [groups.build_config.dockerfile_extensions]
     # deal with dependency changes in master until we create the new vx.y.z instance
-    pre_build          = "RUN cd ${PLAN_DIR} && go mod download github.com/libp2p/go-libp2p && go mod tidy"
+    pre_build = """
+RUN cd ${PLAN_DIR} && \
+    go mod download github.com/libp2p/go-libp2p && \
+    go mod tidy -compat=1.18
```
Comment (marten-seemann) [inline ping/_compositions/go-cross-versions.toml:108]: Why is this needed? go-libp2p should build with Go 1.17.

Comment (laurentsenta) [reply ping/_compositions/go-cross-versions.toml:108]: I thought we wanted to test the current/latest version: https://github.com/libp2p/test-plans/pull/23#discussion_r911688297

Should we test master with 1.18 or 1.17?

**Review by marten-seemann — APPROVED** [review]

File: ping/_compositions/go-cross-versions.toml (hunk 1 above)
Comment (laurentsenta) [reply ping/_compositions/go-cross-versions.toml:108]: (sync'd over Slack, using 1.18 was correct)
